diff --git a/radeon_device.c b/radeon_device.c
index 9e35b14..a785329 100644
--- a/radeon_device.c
+++ b/radeon_device.c
@@ -1635,7 +1635,11 @@ int radeon_suspend_kms(struct drm_device *dev, bool suspend,
 	}
 
 	if (notify_clients)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0)
+		drm_client_dev_suspend(dev);
+#else
 		drm_client_dev_suspend(dev, false);
+#endif
 
 	return 0;
 }
@@ -1739,7 +1743,11 @@ int radeon_resume_kms(struct drm_device *dev, bool resume, bool notify_clients)
 		radeon_pm_compute_clocks(rdev);
 
 	if (notify_clients)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0)
+		drm_client_dev_resume(dev);
+#else
 		drm_client_dev_resume(dev, false);
+#endif
 
 	return 0;
 }
