diff --git a/radeon_fbdev.c b/radeon_fbdev.c
index dc81b0c..38617d8 100644
--- a/radeon_fbdev.c
+++ b/radeon_fbdev.c
@@ -206,7 +206,11 @@ int radeon_fbdev_driver_fbdev_probe(struct drm_fb_helper *fb_helper,
 	struct radeon_device *rdev = fb_helper->dev->dev_private;
 	const struct drm_format_info *format_info;
 	struct drm_mode_fb_cmd2 mode_cmd = { };
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 0, 0)
+	struct fb_info *info = fb_helper->info;
+#else
 	struct fb_info *info;
+#endif
 	struct drm_gem_object *gobj;
 	struct radeon_bo *rbo;
 	struct drm_framebuffer *fb;
@@ -247,12 +251,14 @@ int radeon_fbdev_driver_fbdev_probe(struct drm_fb_helper *fb_helper,
 	fb_helper->funcs = &radeon_fbdev_fb_helper_funcs;
 	fb_helper->fb = fb;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(7, 0, 0)
 	/* okay we have an object now allocate the framebuffer */
 	info = drm_fb_helper_alloc_info(fb_helper);
 	if (IS_ERR(info)) {
 		ret = PTR_ERR(info);
 		goto err_drm_framebuffer_unregister_private;
 	}
+#endif
 
 	info->fbops = &radeon_fbdev_fb_ops;
 
@@ -279,10 +285,12 @@ int radeon_fbdev_driver_fbdev_probe(struct drm_fb_helper *fb_helper,
 
 	return 0;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(7, 0, 0)
 err_drm_framebuffer_unregister_private:
 	fb_helper->fb = NULL;
 	drm_framebuffer_unregister_private(fb);
 	drm_framebuffer_cleanup(fb);
+#endif
 err_kfree:
 	kfree(fb);
 err_radeon_fbdev_destroy_pinned_object:
